src/test/java/org/springframework/data/repository/query/ExtensionAwareEvaluationContextProviderUnitTests.java

```diff
@@ -219,6 +222,87 @@ public CustomExtensionRootObject1 getRootObject() {
 		assertThat(counter.get()).isEqualTo(2);
 	}
 
+	@Test // DATACMNS-1026
+	public void overloadedMethodsGetResolved() throws Exception {
+
+		provider = new ExtensionAwareEvaluationContextProvider(
```

Comment:
This is just in a private class anyway. A separate interface sounds like overkill to me in this case.